### ISSUE TITLE
Similar gyp error

### ISSUE THREAD

**Issue by ProductOfAmerica** [kelektiv/node.bcrypt.js#668]
Windows version: Windows 10 Education v. 1803
NodeJS version: v8.12.0
Yarn version: v0.27.5
Python version: 3.4.2

Error is similar to [here](https://github.com/kelektiv/node.bcrypt.js/issues/476) however is **not** the same.

```
C:\Users\username\WebstormProjects\material-server>yarn install
yarn install v0.27.5
warning ..\..\package.json: No license field
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[1/1] ⠄ bcrypt: node@8.12.0 | win32 | x64
[-/1] ⠄ waiting...
[-/1] ⠄ waiting...
[-/1] ⠄ waiting...
error C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt: Command failed.
Exit code: 1
Command: C:\WINDOWS\system32\cmd.exe
Arguments: /d /s /c node-pre-gyp install --fallback-to-build
Directory: C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt
Output:
node-pre-gyp info it worked if it ends with ok
node-pre-gyp info using node-pre-gyp@0.11.0
node-pre-gyp info using node@8.12.0 | win32 | x64
node-pre-gyp WARN Using needle for node-pre-gyp https download
node-pre-gyp info check checked for "C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt\lib\binding\bcrypt_lib.node" (not found)
node-pre-gyp http GET https://github.com/kelektiv/node.bcrypt.js/releases/download/v3.0.2/bcrypt_lib-v3.0.2-node-v57-win32-x64-unknown.tar.gz
node-pre-gyp http 302 https://github.com/kelektiv/node.bcrypt.js/releases/download/v3.0.2/bcrypt_lib-v3.0.2-node-v57-win32-x64-unknown.tar.gz
node-pre-gyp WARN Pre-built binaries not installable for bcrypt@3.0.2 and node@8.12.0 (node-v57 ABI, unknown) (falling back to source compile with node-gyp)
node-pre-gyp WARN Hit error Remote end closed socket abruptly.

C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" clean )  else (node "" clean )
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@8.12.0 | win32 | x64
gyp info ok

C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" configure --fallback-to-build --module=C:\Users\username\
WebstormProjects\material-server\node_modules\bcrypt\lib\binding\bcrypt_lib.node --module_name=bcrypt_lib --module_path=C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt\lib\binding --napi_version=3 --node_abi_napi=napi --napi_build_version=0 --nod
e_napi_label=node-v57 )  else (node "" configure --fallback-to-build --module=C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt\lib\binding\bcrypt_lib.node --module_name=bcrypt_lib --module_path=C:\Users\username\WebstormProjects\material-server\node_mo
dules\bcrypt\lib\binding --napi_version=3 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v57 )
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@8.12.0 | win32 | x64
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Python34\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                                ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:276:12)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:915:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=C:\\Users\\username\\WebstormProjects\\material-server\\node_modules\\bcrypt\\lib
\\binding\\bcrypt_lib.node" "--module_name=bcrypt_lib" "--module_path=C:\\Users\\username\\WebstormProjects\\material-server\\node_modules\\bcrypt\\lib\\binding" "--napi_version=3" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v57"
gyp ERR! cwd C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt
gyp ERR! node -v v8.12.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'node-gyp.cmd configure --fallback-to-build --module=C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt\lib\binding\bcrypt_lib.node --module_name=bcrypt_lib --module_path=C:\Users\username\WebstormProjects
\material-server\node_modules\bcrypt\lib\binding --napi_version=3 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v57' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at emitTwo (events.js:126:13)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:915:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
node-pre-gyp ERR! System Windows_NT 10.0.17134
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\username\\WebstormProjects\\material-server\\node_modules\\bcrypt\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt
node-pre-gyp ERR! node -v v8.12.0
node-pre-gyp ERR! node-pre-gyp -v v0.11.0
node-pre-gyp ERR! not ok
Failed to execute 'node-gyp.cmd configure --fallback-to-build --module=C:\Users\username\WebstormProjects\material-server\node_modules\bcrypt\lib\binding\bcrypt_lib.node --module_name=bcrypt_lib --module_path=C:\Users\username\WebstormProjects\material-server\node_modules\b
crypt\lib\binding --napi_version=3 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v57' (1)
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by agathver** [comment]
It's a known error with node-pre-gyp which uses needle to download from https and for some reason it doesnt work reliably. 

Install the request module and npm rebuild bcrypt. node-pre-gyp will use request when available.

**Comment by springmeyer** [comment]
Yes using request will fix. Also the root cause is a bad interaction of needle and node v8, which is fixed in https://github.com/tomas/needle/pull/263 and available in needed >= 2.2.4. per https://github.com/tomas/needle/issues/261#issuecomment-423764250

**Comment by tw0517tw** [comment]
node-gyp doesn't support python 3. You need to use python 2 for node-gyp.
